### PR TITLE
Stock API access upgrades

### PR DESCRIPTION
**Pull request by Spaghetti-12**
Added more methods and classes to the APIInterface package to make it easier to implement other features throughout the program.

Added request and response models for a bulk API request, and added a function for it to StockAPIAccess. This should be helpful for implementing the Portfolio method to update all of its stock prices, since now it won't have to make a bunch of individual API calls.

Also modified StockAPIResponse to store historical stock data as well as current price; StockAPIAccess.getPrice now also returns historical data. This will be important if we were going to have historical data show up when we search for a stock.

### CODE REVIEW

**Review by Peaceful36 — APPROVED** [review]
Seems good to me

File: src/main/java/APIInterface/BulkStockAPIResponse.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package APIInterface;
+
+import java.util.HashMap;
+
+public class BulkStockAPIResponse {
+    /**
+     * Response model class containing output from a bulk price request to the
+     * stock price API.
+     */
+    private HashMap<String, Double> symbolToPrice;
```
Comment (eladperlman) [inline src/main/java/APIInterface/BulkStockAPIResponse.java:10]: Should be type map as HashMap is just a specific implementation

File: src/main/java/APIInterface/BulkStockAPIResponse.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (eladperlman) [inline src/main/java/APIInterface/BulkStockAPIResponse.java:11]: same here

File: src/main/java/APIInterface/BulkStockAPIResponse.java (hunk 1 above)
Comment (Spaghetti-12) [reply src/main/java/APIInterface/BulkStockAPIResponse.java]: Fixed!

**Review by inaam10-tech — APPROVED** [review]
Seems good.

**Review by noel-toms — APPROVED** [review]
Looks good.

**Review by Rohh1 — APPROVED** [review]
Change in function names better represent what the function does.

**Pull request merged**